travis/setup_dependencies_common.sh

```diff
@@ -357,14 +360,37 @@ elif [[ $NUMPY_VERSION == pre* ]]; then
         travis_terminate 0
     fi
 elif [[ ! -z $NUMPY_VERSION ]]; then
-    retry_on_known_error conda install $QUIET --no-pin $PYTHON_OPTION numpy=$NUMPY_VERSION $MKL
     export NUMPY_OPTION="numpy=$NUMPY_VERSION"
-    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION numpy=$NUMPY_VERSION $MKL"
+    export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION $NUMPY_OPTION $MKL"
+    NUMPY_INSTALL="conda install $QUIET --no-pin $PYTHON_OPTION $NUMPY_OPTION $MKL"
+    
 else
     export NUMPY_OPTION=""
     export CONDA_INSTALL="conda install $QUIET $PYTHON_OPTION $MKL"
 fi
 
+# try to install numpy:
+if [[ ! -z $NUMPY_INSTALL ]]; then
+    retry_on_known_error $NUMPY_INSTALL || { \
+        if [[ -z $NUMPY_OPTION ]]; then
+            PIP_NUMPY_OPTION="numpy"
+        else
+            if [[ $NUMPY_OPTION =~ ^numpy$is_eq_float$ ]]; then
```

Comment:
so you forbid here to install 1.15.2?

Reply:
You're right, I forgot to handle cases with exact version requirements further below.
This particular check is to add a a wildcard for pip so that it behaves like conda, i.e., to transform the conda requirement `numpy=1.16` into the corresponding pip requirement `numpy==1.16.*`.
I'll add an additional check for arbitrary versions.

Reply:
we already have some workarounds for those when falling back to pip install for conda dependency packages. Maybe simply adding numpy to the CONDA_DEPENDENCIES list would work (and probably should have done a long time ago, but this whole pile of mess just evolved very organically hacking the hacks of an early hack).

Reply:
(numpy originally was handled separately as that was the only dependency astropy had. then we added layers and layers of stuff on top while should have started from the beginning again. But then, now, given that we can't really see what happens to travis in the near future probably it's not worth spending too much time with big refactoring.)

Reply:
I'm (unfortunately) very familiar with pip fallbacks... :wink: 
Using the same functionality here is not an option, though, since then, one would have to disallow numpy installs via conda, leading to a massive amount of pip fallbacks for the rest of the `CONDA_DEPENDENCIES`.
Adding numpy to `CONDA_DEPENDENCIES` will cause single conda installs for all packages as well because the conda install with all packages at once *will* fail. One-by-one installs are a (very) sub-optimal solution since package (inter-)dependencies are then resolved incrementally, which is dead slow...
So, I think that once we know that numpy can't be installed via conda, there is no reason to not add numpy to `PIP_DEPENDENCIES`.

Reply:
at some point I think we need to also check whether there is anything in CONDA_DEPENDENCIES, and if not, but there is PIP_DEPENDENCIES, then just install numpy from pip (we already have a few packages where we switched to pip only installs as it's way faster). Anyway, it's into the weeds for this PR, but maybe something for the future.